### ISSUE TITLE
Discrepancy Between Scaling Factors in Paper and Code, and Typical Loss Magnitudes

### ISSUE THREAD

**Issue by Peterande** [NVlabs/rcm#8]
@zhengkw18 @qsh-zh Excellent work! Could you clarify why the scaling factor reported in the paper is 1.0 and 0.01, whereas in the code the scales for SCM and DMD are set to 100.0 and 1.0, respectively? Additionally, what are the typical magnitudes of the various loss terms during training for text-to-image (T2I) and text-to-video (T2V) models?

**Comment by zhengkw18** [comment]
In our experiments it will not. We find `sum+relative weight 0.01` just generalizes. If using mean for DMD, the quality will be bad on video. Intuitively, DMD provides a large correction force to sCM. But as long as the sCM loss exists (which is mode-covering), it will enforce large penalties if the student does not cover any training data and help maintain diversity. As we said in the paper, the two loss terms operate on different generation paths and do not conflict.

For sCM, using sum will result in a final loss value of 1 because of tangent normalization.

These settings may not fit people's intuition, but are empirically verified by us with over 100k GPU hours of experiments. All we can tell is the final conclusion that `sum+relative weight 0.01` generalizes for T2I and T2V. The concrete analyses of training dynamics and explanations are left for future work.

**Comment by Peterande** [comment]
> In our experiments it will not. We find `sum+relative weight 0.01` just generalizes. If using mean for DMD, the quality will be bad on video. Intuitively, DMD provides a large correction force to sCM. But as long as the sCM loss exists (which is mode-covering), it will enforce large penalties if the student does not cover any training data and help maintain diversity. As we said in the paper, the two loss terms operate on different generation paths and do not conflict.
> 
> For sCM, using sum will result in a final loss value of 1 because of tangent normalization.
> 
> These settings may not fit people's intuition, but are empirically verified by us with over 100k GPU hours of experiments. All we can tell is the final conclusion that `sum+relative weight 0.01` generalizes for T2I and T2V. The concrete analyses of training dynamics and explanations are left for future work.

This aligns well with the gradient direction of DMD—particularly, its magnitude needs to be weighted according to the number of frames, so using sum is indeed reasonable. Thank you very much for your clarification!

**Comment by Peterande** [comment]
> > Using (100, 1) instead of (1, 0.01) is inherited from previous works that operate on FP16, which may better prevent numerical underflow. In our BF16 setting, (100, 1) should behave exactly the same as (1, 0.01).
> 
> Ideally with perfect numerical accuracy, (100, 1) should behave exactly the same as (1, 0.01). In practice. manually loss scaling may still matters thanks to finite bit float number representation and eps in adam.

That sounds very reasonable—thank you! I’ll run some related validations on my end. 😄

**Comment by Peterande** [comment]
> > [@zhengkw18](https://github.com/zhengkw18) I'm glad to hear that the sCM loss still has a significant impact on performance. From some gradient quantification analyses I've done, there is indeed still a noticeable gap between the gradients contributed by the sCM loss and the DMD loss. However, compared to the huge difference in their loss magnitudes, the discrepancy in their gradients is much smaller. This suggests that, despite its much smaller magnitude relative to the DMD loss, the sCM loss still provides non-negligible gradient contributions. Your response has likely clarified many concerns that first-time readers might have had. Thank you very much!
> > Additionally, I’d like to ask you another question: suppose we design a third loss that is entirely unrelated to the current visual generation task—for example, in a multimodal setting where our model can also generate audio. In the context of CM acceleration, this new loss is combined with the CM loss at a 1:1 ratio; similarly, in DMD acceleration, it’s combined with the DMD loss at a 1:1 ratio as well. However, given the current large magnitude discrepancy between rCM’s CM loss and DMD loss, how should I scale this third loss appropriately?
> 
> Yes, I remember regarding the gradient norm, DMD may be around 10x larger than sCM on video. But if we decrease the DMD weight by 10x, the quality will be bad. A large DMD correction force seems necessary for good quality.
> 
> Our finding of the relative loss weight is only verified on single modality and joint training. If you use them separately or incorporate other loss terms, I cannot say exactly how the weight should be set. The best way is to conduct some hyperparameter search and draw conclusions from experiments. If you use sCM and DMD jointly, their relative weight can be determined by observing the quality and diversity.

Understood. Thank you very much for your patient reply!

**Comment by zhengkw18** [comment]
@Peterande We conduct an ablation study on $\lambda$ recently, for your reference

<img width="1540" height="631" alt="Image" src="https://github.com/user-attachments/assets/51270be9-1b47-4e4b-999f-aff785d0bd9d" />
